apps/server/src/shared/domain/entity/task.entity.ts

```diff
@@ -120,58 +162,48 @@ export class Task extends BaseEntityWithTimestamps implements ILearnroomElement,
 		if (this.availableDate && this.availableDate > new Date(Date.now())) {
 			return true;
 		}
-		return false;
-	}
 
-	private getSubmissionItems(): Submission[] {
-		if (!this.submissions.isInitialized(true)) {
-			throw new Error('Submissions items are not loaded.');
-		}
-		const submissions = this.submissions.getItems();
-		return submissions;
+		return false;
 	}
 
-	getSubmittedUserIds(): EntityId[] {
-		const submittedUserIds = this.getSubmissionItems().map((submission) => submission.student.id);
-		const uniqueSubmittedUserIds = [...new Set(submittedUserIds)];
+	// not complet valid, it is based on different parameters and can not be calculated on this place
+	private getMaxSubmissions(): number {
+		const numberOfStudents = this.getUserIds().length;
 
-		return uniqueSubmittedUserIds;
+		return numberOfStudents;
 	}
 
-	getNumberOfSubmittedUsers(): number {
-		const submittedUserIds = this.getSubmittedUserIds();
-		const count = submittedUserIds.length;
+	private getSubmittedSubmissions(): Submission[] {
+		const submissions = this.getSubmissionItems();
+		const gradedSubmissions = submissions.filter((submission) => submission.isSubmitted());
 
-		return count;
+		return gradedSubmissions;
 	}
 
-	getGradedUserIds(): EntityId[] {
-		const gradedUserIds = this.getSubmissionItems()
-			.filter((submission) => submission.isGraded())
-			.map((submission) => submission.student.id);
-		const uniqueGradedUserIds = [...new Set(gradedUserIds)];
+	private getGradedSubmissions(): Submission[] {
+		const submissions = this.getSubmissionItems();
+		const gradedSubmissions = submissions.filter((submission) => submission.isGraded());
 
-		return uniqueGradedUserIds;
+		return gradedSubmissions;
 	}
 
-	getNumberOfGradedUsers(): number {
-		const gradedUserIds = this.getGradedUserIds();
-		const count = gradedUserIds.length;
+	private isSubmittedForUser(user: User): boolean {
+		const submissions = this.getSubmissionItems();
+		const isSubmitted = submissions.some((submission) => submission.isSubmittedForUser(user));
 
-		return count;
+		return isSubmitted;
 	}
 
-	// attention based on this parent use this.getParent() instant
-	getMaxSubmissions(): number {
-		// hack until parents are defined
-		const numberOfStudents = this.course ? this.course.getNumberOfStudents() : 0;
+	private isGradedForUser(user: User): boolean {
+		const submissions = this.getSubmissionItems();
+		const graded = submissions.some((submission) => submission.isGradedForUser(user));
 
-		return numberOfStudents;
+		return graded;
 	}
 
 	createTeacherStatusForUser(user: User): ITaskStatus {
-		const submitted = this.getNumberOfSubmittedUsers();
-		const graded = this.getNumberOfGradedUsers();
+		const submitted = this.getSubmittedSubmissions().length;
```

Comment:
It is possible to calculate it for each user and get the valid count based on getMaxSubmissions() but i am not sure if it is the information that the teachers expect. If they read 10 / 30 submissions are ready, but only 2 submissions exist (because team-submissions with 5 members are allowed) it can be confused, or?
Must be discuss with team and UX.

The getMaxSubmissions count is always based on max team members.

Reply:
Ok if i understand you correct, i should switch this to totalSubmittedUsers / maxUsersOfParent.
The totalSubmittedUsers can be calculated from teamMemberIds and creator in the submission. 

Result in the teacher see how many students has make the work done, but the submission count and totalSubmittedUsers count can be different, if the teacher go to the detail page. 

This is also my favorit, it skip the problem with maxUserOfParent vs maxSubmissions. Only the code is a litte bit harder to understand.

Reply:
guess we need to talk about that personally to make sure we talk about the same details

Reply:
I implemented it on base of the total users that are graded, submitted. This enforce many many more code, but work similar like before. But if you like we can speak.